File: app/src/main/java/com/jinhanexample/mvvmSample/java/author/ui/AuthorViewModel.java
package com.jinhanexample.mvvmSample.java.author.ui;

import android.app.Application;
import android.util.Log;

import androidx.lifecycle.AndroidViewModel;
import androidx.lifecycle.LiveData;
import androidx.lifecycle.ViewModel;

import com.jinhanexample.mvvmSample.java.author.data.AuthorEntity;
import com.jinhanexample.mvvmSample.java.author.data.AuthorRepository;

import java.util.List;

public class AuthorViewModel extends AndroidViewModel {

    private AuthorRepository repository;
    private LiveData<List<AuthorEntity>> authorData;

    public AuthorViewModel(Application application) {
        super(application);
        this.repository = new AuthorRepository(application);
        authorData = repository.getAll();
    }

    LiveData<List<AuthorEntity>> getAll() {
        return this.authorData;
    }

    void insert(AuthorEntity authorEntity){
        this.repository.insert(authorEntity);
    }

    void delete(AuthorEntity authorEntity){
        this.repository.delete(authorEntity);
    }


}

File: app/src/main/java/com/jinhanexample/mvvmSample/kotlin/author/data/AuthorViewModel.kt
package com.jinhanexample.mvvmSample.kotlin.author.data

import android.app.Application
import androidx.lifecycle.AndroidViewModel
import androidx.lifecycle.LiveData


// AuthorRepository에 접근하여 데이터를 불러온다.
class AuthorViewModel(application: Application) : AndroidViewModel(application) {

    // AuthorRepository 인스턴스 생성
    private val repository = AuthorRepository(application = application)

    //authors 데이터 가져오기
    private val authors = repository.getAll()

    fun getAll() : LiveData<List<AuthorEntity>>{
        return this.authors
    }

    //repository에 접근하여 insert와 delete기능 수행

    fun insert(authorEntity: AuthorEntity){
        repository.insert(authorEntity)
    }

    fun delete(authorEntity: AuthorEntity){
        repository.delete(authorEntity)
    }
}